pkg/controller/resourcegraphdefinition/controller.go

```diff
@@ -100,7 +105,50 @@ func (r *ResourceGraphDefinitionReconciler) SetupWithManager(mgr ctrl.Manager) e
 				MaxConcurrentReconciles: r.maxConcurrentReconciles,
 			},
 		).
-		Complete(reconcile.AsReconciler[*v1alpha1.ResourceGraphDefinition](mgr.GetClient(), r))
+		Watches(
+			&extv1.CustomResourceDefinition{},
+			handler.EnqueueRequestsFromMapFunc(r.findRGDsForCRD),
+			builder.WithPredicates(predicate.Funcs{
+				UpdateFunc: func(e event.UpdateEvent) bool {
+					return true
+				},
+				CreateFunc: func(e event.CreateEvent) bool {
+					return false
+				},
+				DeleteFunc: func(e event.DeleteEvent) bool {
+					return false
+				},
+			}),
+		).
+		Complete(reconcile.AsReconciler(mgr.GetClient(), r))
+}
+
+// findRGDsForCRD returns a list of reconcile requests for the ResourceGraphDefinition
+// that owns the given CRD. It is used to trigger reconciliation when a CRD is updated.
+func (r *ResourceGraphDefinitionReconciler) findRGDsForCRD(ctx context.Context, obj client.Object) []reconcile.Request {
+	crd, ok := obj.(*extv1.CustomResourceDefinition)
+	if !ok {
+		return nil
+	}
+
+	// Check if the CRD is owned by a ResourceGraphDefinition
+	if !metadata.IsKROOwned(crd.ObjectMeta) {
+		return nil
+	}
+
+	rgdName, ok := crd.Labels[metadata.ResourceGraphDefinitionNameLabel]
```

Comment:
This is easy to defeat. An admin can change/remove the label value and effectively unmanage that CRD temporarily. And then change the CRD.